frontend/src/app/party/party-payment/party-payment.component.ts

```diff
@@ -7,9 +10,58 @@ import { Component, OnInit } from '@angular/core';
 })
 export class PartyPaymentComponent implements OnInit {
 
-  constructor() { }
+  @Input() party: Party;
+  @Input() partyState: PartyState;
+  @Input() user: User;
+  @Input() menus: Menu[];
+  @Output() toNextState: EventEmitter<void> = new EventEmitter();
+
+  myMenus: [number, number, number][]; // menuId, menuPrice, menuQuantity
+  totalCost: number;
+
+  constructor() {
+    this.toNextState = new EventEmitter();
+  }
 
   ngOnInit() {
+    if (this.partyState === undefined) {
```

Comment:
Rather exit directly if party state from Input directive is undefined, keep track of the party state with `ngOnChange` and update the `myMenus` on every change.